Setup production

garrickb opened the pull request.

Commit:
Set up TravisCI testing.

Commit:
Fix deployment. Add to README.

Commit:
Build the application in travis ci.

Commit:
fix tslint issue?

Commit:
Remove unused ts import.

Commit:
Fix capitalization of import.

Commit:
Deploy to GitHub Pages

Commit:
Fix ghpages issue.

Commit:
Move where index is copied.

Commit:
Change `git add` command.

Commit:
Add ; to each line.

Commit:
Change API url to reflect if we're in dev or prod.

Commit:
Change JS location temporarily to test gh-pages.

Commit:
Use HTTPS.

jackhorton commented on src/constants.tsx:
```diff
@@ -5,8 +5,8 @@ if (__DEVTOOLS__) {
-    AUTH_API_URL = "http://identity.gmbuddy.com";
-    API_URL = "http://api.gmbuddy.com";
+    AUTH_API_URL = "https://identity.gmbuddy.com";
+    API_URL = "https://api.gmbuddy.com";
```
Does this work? Im pretty sure this doesnt work.

jackhorton replied:
I certainly dont have https set up

jackhorton committed:
Create circle.yml

jackhorton committed:
Update circle.yml

jackhorton committed:
Update circle.yml

jackhorton committed:
Update circle.yml

jackhorton committed:
Update circle.yml

jackhorton committed:
Create circle-deploy.sh

jackhorton committed:
Update circle.yml

jackhorton committed:
Update index.html

Commit:
Don't use HTTPS since it's not setup.

Commit:
Remove travisci.

Commit:
Remove travis script. Add circle badge to README.

Commit:
Only do stuff on master.

Commit:
lol

Merge branch 'master' of https://github.com/GMBuddy/gmbuddy-web into setup-production

# Conflicts:
#	src/campaign/actions/creator/thunks.tsx
#	src/campaign/actions/fetch/actions.tsx
#	src/campaign/actions/fetch/thunks.tsx
#	src/campaign/actions/reducer.tsx
#	src/campaign/component/creator/CampaignInvite.tsx
#	src/campaign/create/components/CampaignDesigner.tsx
#	src/campaign/create/components/CampaignReview.tsx
#	src/campaign/create/containers/CampaignCreator.tsx
#	src/campaign/fetch/components/FetchCampaign.tsx
#	src/campaign/fetch/components/FetchCampaigns.tsx
#	src/campaign/fetch/containers/CampaignView.tsx
#	src/campaign/fetch/containers/CampaignsView.tsx
#	src/campaign/index.scss
#	src/campaign/routes.tsx
#	src/campaign/view/components/Campaign.tsx
#	src/campaign/view/components/Campaigns.tsx
#	src/character-creator/index.scss
#	src/character-creator/routes.tsx
#	src/character-data/dnd35/CharacterData.tsx
#	src/character-viewer/dnd35/CharacterSkills.tsx
#	src/character-viewer/dnd35/index.scss
#	src/character-viewer/index.scss
#	src/character-viewer/routes.tsx
#	src/character/creator/components/CharacterModule.tsx
#	src/character/creator/components/CharacterStepButtons.tsx
#	src/character/creator/components/CharacterStepper.tsx
#	src/character/creator/components/dnd35/CharacterDetails.tsx
#	src/character/creator/components/dnd35/CharacterItemModal.tsx
#	src/character/creator/components/dnd35/CharacterItems.tsx
#	src/character/creator/components/dnd35/CharacterReview.tsx
#	src/character/creator/components/shared/CharacterStats.tsx
#	src/character/creator/containers/CharacterCreator.tsx
#	src/character/creator/containers/DnD35CharacterCreator.tsx
#	src/character/creator/containers/NotFoundCharacterCreator.tsx
#	src/character/creator/styles/dnd35.scss
#	src/character/viewer/components/dnd35/CharacterView.tsx
#	src/character/viewer/components/dnd35/DetailsPage.tsx
#	src/character/viewer/components/dnd35/DnD35CharacterViewer.tsx
#	src/character/viewer/components/dnd35/ItemsPage.tsx
#	src/character/viewer/components/dnd35/SkillsPage.tsx
#	src/character/viewer/components/dnd35/StatsPage.tsx
#	src/character/viewer/containers/CharacterViewer.tsx
#	src/layout/components/Sidebar.tsx
#	src/main.scss
#	src/reducer.tsx
#	src/routes.tsx
#	typings/custom/gmbuddy.d.ts
